vulture_os/toolkit/api_parser/cisco_umbrella_managed_org/cisco_umbrella_managed_org.py

```diff
@@ -0,0 +1,269 @@
+                try:
+                    self._get_customer_token(customer_id)
+                except Exception as e:
+                    logger.error(f"[{__parser__}]:execute: {e}", extra={'frontend': str(self.frontend)})
+            access_token = self.cisco_umbrella_managed_org_customers_tokens[customer_id]["access_token"]
+            self._connect(access_token)
+            for log_type in log_types:
+                timestamp_field_name = f"app{customer_id}_{log_type}"
+                since = self.last_collected_timestamps.get(timestamp_field_name) or (timezone.now() - timedelta(days=1))
+                to = min(timezone.now(), since + timedelta(hours=24))
+                logger.info(f"[{__parser__}]:execute: getting {log_type} logs from {since} to {to}, for organization {customer_id}", extra={'frontend': str(self.frontend)})
+                try:
+                    logs = self.get_logs(customer_id, log_type, since, to)
+                except Exception as e:
+                    logger.error(f"[{__parser__}]:execute: {e}", extra={'frontend': str(self.frontend)})
+                index = 0
+                logs_count = self.LIMIT_MAX
+                while logs_count == self.LIMIT_MAX and index <= self.OFFSET_MAX:
+                    try:
+                        response = self.get_logs(customer_id, log_type, since, to)
+                    except Exception as e:
+                        logger.error(f"[{__parser__}]:execute: {e}", extra={'frontend': str(self.frontend)})
+                    # Downloading may take some while, so refresh token in Redis
+                    self.update_lock()
+                    logs = response['data']
+                    logs_count = len(logs)
+                    logger.info(f"[{__parser__}]:execute: got {logs_count} lines", extra={'frontend': str(self.frontend)})
+                    index += logs_count
+                    logger.info(f"[{__parser__}]:execute: retrieved {index} lines", extra={'frontend': str(self.frontend)})
+                    self.write_to_file([self.format_log(log) for log in logs])
+                    # Writting may take some while, so refresh token in Redis
+                    self.update_lock()
+                # When there are more than 15000 logs, last_api_call is the timestamp of the last log
+                if logs_count == self.LIMIT_MAX and index == self.OFFSET_MAX + self.LIMIT_MAX:
+                    timestamp = logs[-1]['timestamp']/1000
+                    self.last_collected_timestamps[timestamp_field_name] = datetime.fromtimestamp(timestamp, tz=timezone.now().astimezone().tzinfo)
+                else:
+                    # All logs have been recovered, the parser can be stopped
+                    self.last_collected_timestamps[timestamp_field_name] = to
+                    break
+                self.frontend.save()
```

Comment:
```suggestion
                if logs_count == self.LIMIT_MAX and index == self.OFFSET_MAX + self.LIMIT_MAX:
                    timestamp = logs[-1]['timestamp']/1000
                    self.last_collected_timestamps[timestamp_field_name] = datetime.fromtimestamp(timestamp, tz=timezone.now().astimezone().tzinfo)
                elif index > 0:  
                    # All logs have been recovered, the parser can be stopped  
                    self.last_collected_timestamps[timestamp_field_name] = to  
                    break  
                elif since < timezone.now() - timedelta(hours=24):  
                    self.last_collected_timestamps[timestamp_field_name] = since + timedelta(hours=1)
                self.frontend.last_collected_timestamps = self.last_collected_timestamps
                self.frontend.save()
```